internal/store/driver.go

```diff
@@ -642,8 +754,38 @@ func (d *Driver) calculateHTTPSEdges() map[string]ingressv1alpha1.HTTPSEdge {
 			edgeMap[rule.Host] = edge
 		}
 	}
+}
 
-	return edgeMap
+func (d *Driver) calculateHTTPSEdgesFromGateway(edgeMap map[string]ingressv1alpha1.HTTPSEdge) {
+	gateways := d.store.ListGateways()
+
+	// get ingress domains since we don't want gateway to interact with them
+	ingressDomains := make(map[string]ingressv1alpha1.Domain)
```

Comment:
I think this should happen in advance, as part of calculate domains? 🤔

Reply:
Yes, this already happens in calculate domains, but since there's a risk of a domain name collision the driver need to know which domains belong to ingress while doing calculations for the gateway.

Reply:
I meant that we can return this/the error while `calculateDomainsFromGateway`, so domain can be only part of `Ingress` or `Gateway`

Reply:
updated here: https://github.com/ngrok/kubernetes-ingress-controller/pull/343/commits/dfbc8c4d59f803261b92a9c165620e968d2443c6